datepicker.js

```diff
@@ -197,53 +199,26 @@ class DatePicker extends Component {
     }, 200);
   }
 
-  onDatePicked({action, year, month, day}) {
-    if (action !== DatePickerAndroid.dismissedAction) {
-      this.setState({
-        date: new Date(year, month, day)
-      });
-      this.datePicked();
-    } else {
+  onDatetimePicked(event, date) {
+    const {mode} = this.props;
+    const {androidDatePickerMode} = this.state;
+    // cancel pressed
+    if (date === undefined || date === null) {
+      this.setState({androidDatePickerMode: undefined});
       this.onPressCancel();
+      return;
     }
-  }
-
-  onTimePicked({action, hour, minute}) {
-    if (action !== DatePickerAndroid.dismissedAction) {
-      this.setState({
-        date: Moment().hour(hour).minute(minute).toDate()
-      });
-      this.datePicked();
-    } else {
-      this.onPressCancel();
-    }
-  }
-
-  onDatetimePicked({action, year, month, day}) {
-    const {mode, androidMode, format = FORMATS[mode], is24Hour = !format.match(/h|a/)} = this.props;
 
-    if (action !== DatePickerAndroid.dismissedAction) {
-      let timeMoment = Moment(this.state.date);
-
-      TimePickerAndroid.open({
-        hour: timeMoment.hour(),
-        minute: timeMoment.minutes(),
-        is24Hour: is24Hour,
-        mode: androidMode
-      }).then(this.onDatetimeTimePicked.bind(this, year, month, day));
-    } else {
-      this.onPressCancel();
-    }
-  }
+    // confirm pressed
+    const timePickerVisible = androidDatePickerMode === DATE_MODE && mode === DATETIME_MODE;
+    const _androidDatePickerMode = timePickerVisible ? TIME_MODE : undefined;
```

Comment:
Is it safe to have an undefined state i.e. for _androidDatePickerMode?

Reply:
removing the undefined adding a state for androidDatePickerVisible